dotnet/src/Plugins/Plugins.Core/HttpPlugin.cs

```diff
@@ -31,8 +32,9 @@ public HttpPlugin() : this(HttpClientProvider.GetHttpClient())
     /// <remarks>
     /// <see cref="HttpPlugin"/> assumes ownership of the <see cref="HttpClient"/> instance and will dispose it when the plugin is disposed.
     /// </remarks>
-    public HttpPlugin(HttpClient client) =>
-        this._client = client;
+    [ActivatorUtilitiesConstructor]
+    public HttpPlugin(HttpClient? client = null) =>
```

Comment:
Is there any benefit for adding a default value? Wondering that if we had `HttpClient` registed as a service in the serviceProvider, would the client be `null` or be the service we injected?

Reply:
Without a default value, it makes the HttpClient required as part of service resolution, meaning trying to get one of these via DI will throw an exception of there's no HttpClient configured in the container.

Reply:
Should we consider removing the parameter-less constructor because this constructor has a default value, if could be removed, maybe the `ActivatorUtilitiesConstructor` can be removed since there's only one constructor then

Reply:
We could... I kept it so that anyone with a `new()` constraint would still work.